common/tier4_planning_rviz_plugin/include/path/display.hpp

```diff
@@ -82,19 +82,14 @@ class AutowarePathWithDrivableAreaDisplay : public AutowarePathBaseDisplay<T>
 {
 public:
   AutowarePathWithDrivableAreaDisplay()
+  : property_drivable_area_view_{"View Drivable Area", true, "", this},
+    property_drivable_area_alpha_{"Alpha", 0.999, "", &property_drivable_area_view_},
+    property_drivable_area_color_{"Color", QColor(0, 148, 205), "", &property_drivable_area_view_},
+    property_drivable_area_width_{"Width", 0.3f, "", &property_drivable_area_view_}
   {
-    property_drivable_area_view_ = new rviz_common::properties::BoolProperty(
-      "View Drivable Area", true, "", this, SLOT(updateVisualization()));
-    property_drivable_area_alpha_ = new rviz_common::properties::FloatProperty(
-      "Alpha", 0.999, "", property_drivable_area_view_, SLOT(updateVisualization()), this);
-    property_drivable_area_alpha_->setMin(0.0);
-    property_drivable_area_alpha_->setMax(1.0);
-    property_drivable_area_color_ = new rviz_common::properties::ColorProperty(
-      "Color", QColor(0, 148, 205), "", property_drivable_area_view_, SLOT(updateVisualization()),
-      this);
-    property_drivable_area_width_ = new rviz_common::properties::FloatProperty(
-      "Width", 0.3f, "", property_drivable_area_view_, SLOT(updateVisualization()), this);
-    property_drivable_area_width_->setMin(0.001);
+    property_drivable_area_alpha_.setMin(0.0);
+    property_drivable_area_alpha_.setMax(1.0);
+    property_drivable_area_width_.setMin(0.001);
```

Comment:
If `SLOT(updateVisualization()) is removed`, I think visualization is not updated after parameters are changed.

Reply:
(Let me unresolve the comment not to forget the reason not to use `SLOT(updateVisualization)` :pray: )
I had a discussion with Saito-san internally.

- In order to register the `updateVisualization` function as `SLOT`, the class that has the function has `Q_OBJECT` type.
    - FYI: `Error: Class declaration lacks Q_OBJECT macro.`
- The template class cannot have `Q_OBJECT` type.
   - FYI: `Error: Template classes not supported by Q_OBJECT.`
- The derived class from the template class with the `updateVisualiation` function can be compiled, but the runtime error `No such slot rviz_common::_RosTopicDisplay::updateVisualization()` occurs.

In short, by the change to use a template class in order to make path/trajectory/... implementation common, now we cannot use  `SLOT(updateVisualization)`.